packages/constellation/src/components/Compositions/Input/Input.tsx

```diff
@@ -0,0 +1,84 @@
+import classnames from 'classnames'
+import React, { forwardRef, useState } from 'react'
+
+import { IconVisibilityOff, IconVisibilityOn } from '../..'
+import { InputProps } from './Input.types'
+
+const Input = forwardRef<HTMLInputElement, InputProps>(
+  (
+    {
+      disabled,
+      helperText = '',
+      id,
+      label = '',
+      placeholder,
+      prefix,
+      state,
+      type = 'text',
+      ...args
+    },
+    ref,
+  ) => {
+    const [showPassword, setShowPassword] = useState(false)
+    const password = type === 'password'
+
+    return (
+      <div className='constellation'>
+        <label className='text-sm text-title select-none' htmlFor={id}>
+          {label}
+        </label>
+        <div
+          className={classnames(
+            'h-12 rounded-lg flex border border-solid items-center focus-within:border-primary', // TODO remove border-solid when preflight workaround is in
+            {
+              'bg-background-light border-background-light': !disabled,
+              'bg-medium border-medium': disabled,
+              'border-error': state === 'error',
+              'border-success': state === 'success',
+            },
+          )}
+        >
+          {prefix?.length ? (
+            <label className='text-muted text-sm select-none pl-4' htmlFor={id}>
+              {prefix}
+            </label>
+          ) : null}
+          <input
+            ref={ref}
+            {...args}
+            id={id}
+            className={classnames(
+              'text-title placeholder:text-muted h-full pl-4 flex-grow bg-transparent outline-none border-none',
+              { 'text-muted': disabled },
+            )}
+            placeholder={placeholder}
+            disabled={disabled}
+            type={password && !showPassword ? 'password' : 'text'}
+          />
+          {password && ( // TODO: replace with postfix prop
+            <button
+              className='mr-4 pl-2 outline-none border-none bg-transparent flex items-center justify-center cursor-pointer'
+              onClick={() => setShowPassword(!showPassword)}
```

Comment:
```suggestion
              onClick={() => setShowPassword(showPassword => !showPassword)}
```

This makes it not dependent on the showPassword prop, possibly removing one rerender